discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -0,0 +1,352 @@
+    warp::any().map(move || config.clone())
+}
+
+fn with_model(model: Model) -> impl Filter<Extract = (Model,), Error = Infallible> + Clone {
+    warp::any().map(move || model.clone())
+}
+
+fn with_client(client: Client) -> impl Filter<Extract = (Client,), Error = Infallible> + Clone {
+    warp::any().map(move || client.clone())
+}
+
+fn serialize_to_ndjson(articles: &Vec<Article>) -> Result<Bytes, GenericError> {
+    let mut bytes = BytesMut::new();
+
+    fn write_record(article: &Article, bytes: &mut BytesMut) -> Result<(), GenericError> {
+        let bulk_op_instruction = article.create_bulk_op_instruction();
+        let bulk_op_instruction = serde_json::to_vec(&bulk_op_instruction)?;
+        let article_bytes = serde_json::to_vec(article)?;
+
+        bytes.put_slice(&bulk_op_instruction);
+        bytes.put_u8(b'\n');
+        bytes.put_slice(&article_bytes);
+        bytes.put_u8(b'\n');
+        Ok(())
+    }
+
+    for article in articles {
+        write_record(article, &mut bytes)?;
+    }
+
+    Ok(bytes.freeze())
+}
+
+#[derive(Debug)]
+struct TooManyDocumentsError;
+impl Reject for TooManyDocumentsError {}
+
+#[derive(Debug)]
+struct EmbeddingsCalculationError(Vec<String>);
+impl Reject for EmbeddingsCalculationError {}
+
+#[derive(Debug)]
+struct SerializeNdJsonError;
+impl Reject for SerializeNdJsonError {}
+
+#[derive(Debug)]
+struct ElasticOpError;
+impl Reject for ElasticOpError {}
+
+#[derive(Debug)]
+struct ReceivingOpError;
+impl Reject for ReceivingOpError {}
+
+fn handle_elastic_error(err: reqwest::Error) -> Rejection {
+    error!("ElasticOpError {:#?}", err);
+    warp::reject::custom(ElasticOpError)
+}
+
+/// An API error serializable to JSON.
+#[derive(Serialize)]
+struct ErrorMessage {
+    code: u16,
+    message: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    errored_ids: Option<Vec<String>>,
+}
+
+async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible> {
+    let code;
+    let message;
+    let mut errored_ids = None;
+
+    if err.is_not_found() {
+        code = StatusCode::NOT_FOUND;
+        message = "NOT_FOUND";
+    } else if let Some(_) = err.find::<warp::filters::body::BodyDeserializeError>() {
+        code = StatusCode::BAD_REQUEST;
+        message = "REQUEST_BODY_DESERIALIZATION_ERROR";
+    } else if let Some(ElasticOpError) = err.find() {
+        code = StatusCode::BAD_REQUEST;
+        message = "ELASTIC_ERROR";
+    } else if let Some(ReceivingOpError) = err.find() {
+        code = StatusCode::BAD_REQUEST;
+        message = "RECEIVING_OPERATION_ERROR";
+    } else if let Some(SerializeNdJsonError) = err.find() {
+        code = StatusCode::BAD_REQUEST;
+        message = "NDJSON_SERIALIZATION_ERROR";
+    } else if let Some(EmbeddingsCalculationError(ids)) = err.find() {
+        code = StatusCode::UNPROCESSABLE_ENTITY;
+        message = "EMBEDDING_CALCULATION_ERROR";
```

Comment:
This is a bit too specific, the user really don't care. Maybe UNPROCESSABLE_DOCUMENTS
or something like that. We can also log to `error!` because this should not happen so we need to look into it

Reply:
please check again